usaspending_api/download/helpers/elasticsearch_download_functions.py

```diff
@@ -190,3 +198,29 @@ def query(cls, filters: dict, download_job: DownloadJob) -> QuerySet:
         )
 
         return queryset
+
+
+class SubawardsElasticsearchDownload(_ElasticsearchDownload):
+    _source_field = "broker_subaward_id"
+    _filter_query_func = QueryWithFilters.generate_subawards_elasticsearch_query
+    _search_type = SubawardSearch
+
+    @classmethod
+    def query(cls, filters: dict, download_job: DownloadJob) -> QuerySet:
+        filter_options = {}
+        time_period_obj = SubawardSearchTimePeriod(
+            default_end_date=settings.API_MAX_DATE, default_start_date=settings.API_MIN_DATE
+        )
+        filter_options["time_period_obj"] = time_period_obj
+        base_queryset = DBSubawardSearch.objects.all()
+        cls._populate_download_lookups(filters, download_job, **filter_options)
+        queryset = base_queryset.extra(
+            tables=["download_job_lookup"],
+            where=[
+                '"download_job_lookup"."download_job_id" = %s '
+                'AND "download_job_lookup"."lookup_id" = "transaction_search"."transaction_id"'
```

Comment:
Should this be using a Subaward table and ID instead of Transactions?

Reply:
No

Reply:
I took a deeper dive look at this file and set of classes, and I want to push back on this a bit. 

Take a look at the logic of `_populate_download_lookups()`. 

One of the first things it does is create an Elasticsearch search object based on the `_search_type()` then applies filters and requests ids based on the `_source_field`, which is `broker_subaward_id` in this case.

The `_get_download_ids_generator()` executes the search and returns the list of ids. This methods' signature needs to be updated to allow for `SubawardSearch` as a search type. Those ids are stored in the `download_job_lookup` table. 

So going back to my original feedback, if `broker_subaward_id` is the field being stored in `lookup_id`, why would we be joining it to `transaction_id`  on the `transaction_search` table when querying data from `DBSubawardSearch`?

Reply:
This was copied from subawards filter, as it originally was. It shouldn't be using subaward search because that's not how it was originally. I can see that lookup_id might need to be changed. But that's it, let me look into it.

Reply:
Ok fixed.

Reply:
Main change looks good, but `_get_download_ids_generator()` method signature still needs to be updated to include `SubawardSearch`

Reply:
Thank you!